app/Jobs/ImportTurboVotePosts.php

```diff
@@ -94,32 +97,10 @@ public function handle(Rogue $rogue)
                     ]);
 
                     if (! $post['data']) {
-                        $tvCreatedAtMonth = strtolower(Carbon::parse($record['created-at'])->format('F-Y'));
-                        $sourceDetails = isset($referralCodeValues['source_details']) ? $referralCodeValues['source_details'] : null;
-                        $postDetails = $this->extractDetails($record);
-
-                        $postData = [
-                            'campaign_id' => (int) $referralCodeValues['campaign_id'],
-                            'campaign_run_id' => (int) $referralCodeValues['campaign_run_id'],
-                            'northstar_id' => $user->id,
-                            'type' => 'voter-reg',
-                            'action' => $tvCreatedAtMonth . '-turbovote',
-                            'status' => $this->translateStatus($record['voter-registration-status'], $record['voter-registration-method']),
-                            'source' => 'turbovote',
-                            'source_details' => $sourceDetails,
-                            'details' => $postDetails,
-                        ];
-
-                        try {
-                            $post = $rogue->createPost($postData);
-
-                            if ($post['data']) {
-                                $this->stats['countPostCreated']++;
-                            }
-                        } catch (\Exception $e) {
-                            info('There was an error storing the post for: ' . $record['id'], [
-                                'Error' => $e->getMessage(),
-                            ]);
+                        $post = CreateTurboVotePostInRogue::dispatch($record, $referralCodeValues, $user);
+
+                        if ($post) {
```

Comment:
I don't think that's what's happening here. The `dispatch` method allows chaining additional configuration methods, like `CreateTurboVotePostInRogue::dispatch(…)->onQueue('queueName')` by returning a [`PendingDispatch` object](https://github.com/laravel/framework/blob/aac6144db22768c780ab5a19c1441830254c6057/src/Illuminate/Foundation/Bus/Dispatchable.php#L9-L17) which since it's not `null`, will evaluate truthily in your conditional below.

Because we're using a queue job here, we won't be able to see the result of the job synchronously and will have to find another way to tell an admin that the whole set of jobs has completed. (And that's kind of a good thing, since it means this job can "finish" a lot quicker and is less likely to time out! 😄)

Reply:
Yes, this looks great! What we're doing here is splitting the work into the smallest "steps" we can do at once (so first parse through all the rows in the CSV, then one-by-one make the API calls creating each user/post). 

This should help keep things from timing out since the slow part of this whole process is spending X seconds per row to make the API calls to Northstar & Rogue. So for a CSV with 1000 rows: instead of, say, a 1005 second job to read the CSV and process all the rows, we've got one 5 second job to parse the CSV and 1000 one-second jobs to send each row to our backends.

Reply:
For now, let's just remove the stat-tracking in this job since it isn't particularly helpful when this is split into separate jobs (and we can see the same information with a little more work by peeking at the logs for this application).

Reply:
ah ok that makes more sense when thinking about not using the stat, thanks @DFurnes ! I was initially thinking if the first job is dependent on the responses from the smaller jobs, then the total time spent in the first job isn't really cut down any since it still waits for the responses of the other jobs. This makes more sense, especially if we cut the stat-tracking!